### ISSUE TITLE
Preferences dialog

### ISSUE THREAD

**Issue by mgedmin** [gtimelog/gtimelog#47]
There should be a Preferences dialog that lets you set:

- "virtual midnight" (assuming I don't find a way to get rid of this altogether)
- goal for work hours per day
- goal for office hours (see #46)

Ideally that's it.  Perhaps some reporting preferences may need to be specified as well, but I haven't figured how reporting will work in the new UI (#31) yet.

(Originally filed at https://bugs.launchpad.net/gtimelog/+bug/220766)

**Comment by mgedmin** [comment]
Ok, the new UI uses the following settings (described in the [schema](https://github.com/gtimelog/gtimelog/blob/experimental-ui/org.gtimelog.gschema.xml):

- hours
- office-hours
- virtual-midnight
- name
- sender
- list-email (can be edited inline)
- remote-task-list (semi-functional)
- task-list-url (semi-functional)

**Comment by mgedmin** [comment]
I see two groups of controls:

**Data entry**
```
  Virtual midnight       [ 02:00 ]
  Work hours             [  8:00 ] per day
  Office hours           [  9:00 ] (including breaks)
```
**Reports**
```
  Your nickname          [ JRH                            ]
  Your email             [ J. R. Hacker <jrh@example.com> ]
  Send reports to        [ activity@example.com           ]
```

**Comment by mgedmin** [comment]
Ok, I've got a mockup I don't hate at first sight.
![screenshot](https://cloud.githubusercontent.com/assets/159967/9769331/2eea27b8-5732-11e5-91a3-8de7c809b0bb.png)

**Comment by mgedmin** [comment]
I'm semi-blocked by https://bugzilla.gnome.org/show_bug.cgi?id=644927.

Even if that one got fixed upstream, I'd have to find a workaround I can deploy now.  Like not using Gtk.SpinButton for the virtual midnight setting.  (Ideally I'd drop the virtual midnight setting and do without.)

**Comment by kronn** [comment]
You need the virtual midnight setting to group entries by day, right? IIRC, there was the idea to use the blank lines (which are currently optional) as day-markers.

This could also solve problems for people who _sometimes_ work a night-shift.

I think that the data-format is even with this change still simple enough to not endanger interoperability with other tools. (e.g. I create the "start"-event manually when starting my desktop environment)

**Comment by mgedmin** [comment]
I wanted to use blank lines for day grouping, but consider this:

```
# day 1: very simple, date: 2005-01-02
2005-01-02 12:00: started **
2005-01-02 13:00: work

# day 2: still simple, stayed late at work -- 2005-02-01
2005-02-01 21:00 starting late **
2005-02-02 01:00 working late

# day 3: is this 2005-03-01 or 2005-03-02?
2005-03-02 01:00 starting very late **
2005-03-02 03:00 working even later

# day 4: if this is 2005-03-02 then the previous day cannot have been 2005-03-01
2005-03-02 09:00 starting early **
2005-03-02 12:00 working hard
```

How do I parse day 3 unambiguously?

**Comment by mgedmin** [comment]
The other place where virtual_midnight is used is to determine when to add a blank line to the file: when the date/time of the previous entry is on the different side of a virtual_midnight boundary from now.  I suppose I could add a separate button to start a new working day, but hmm?  Some thought is needed.

**Comment by mgedmin** [comment]
Ok, the Preferences dialog exists in the experimental-ui branch, so I'm closing this issue.

There should be a separate issue for possibly maybe getting rid of the virtual-midnight setting.
